framework/src/timeintegrators/ActuallyExplicitEuler.C

```diff
@@ -219,6 +224,32 @@ ActuallyExplicitEuler::solve()
   nonlinear_system.setSolution(*libmesh_system.current_local_solution);
 
   libmesh_system.nonlinear_solver->converged = converged;
+
+  // Compute the final nonlinear residual if specified. Note that there are two
+  // methods for computing the nonlinear residual:
+  //   1. As solved by this time integrator: This would keep the steady-state
+  //      residual as it was already computed and then manually compute and add
+  //      the temporal residual for a given "solve_type" ("consistent",
+  //      "lumped", or "lump_preconditioned"). Caveats include the following:
+  //        * If there are constraints enforced on the system after the solve,
+  //          then there will be an inconsistency between the solution and the
+  //          nonlinear residual, unless the nonlinear residual is also modified
+  //          to account for the constraints.
+  //   2. As indicated by the user: This would recompute the entire nonlinear
+  //      residual after the solution is updated. Caveats include the following:
+  //        * The user must be sure to set "implicit" flags as necessary to
+  //          reproduce the behavior of this time integrator.
+  //        * This does not respect the "solve_type" option selected by the
+  //          user, so if using a lumped mass matrix, the user will need to
+  //          set the lumping parameter in the time kernels.
+  // Currently, Method 2 will be used, since there is consistency with the
```

Comment:
👎 Both of these things: setting the "implicit" flags and manually setting lumping are a no-go.  Goes against the entire ethos of ActuallyExplicitEuler.

Reply:
To be clear, the user only needs to do these if the user is testing the final nonlinear residual. Method 1 has this issue with the constraints inconsistency, and Method 2 is cumbersome but should compute the nonlinear residual consistently. Both suck, but I find Method 1 to be much less useful because the main reason why the final nonlinear residual would not be very small (and thus we'd be testing it) is because of constraints. What do you think we should do? Maybe implement both methods?

Reply:
Nope: need to think harder about it and find the "right" thing to do.  Let's get together and talk about this later today.